src/FuncSharp.Examples/ITry.cs

```diff
@@ -0,0 +1,199 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace FuncSharp.Examples
+{
+    public class ITryUsages
+    {
+        public class ClassWithMultiplePropertiesToValidate
+        {
+            public ClassWithMultiplePropertiesToValidate(decimal multiplicationResult, decimal additionResult, decimal subtractionResult)
+            {
+                MultiplicationResult = multiplicationResult;
+                AdditionResult = additionResult;
+                SubtractionResult = subtractionResult;
+            }
+
+            public decimal MultiplicationResult { get; }
+            public decimal AdditionResult { get; }
+            public decimal SubtractionResult { get; }
+        }
+
+        public enum NetworkOperationError
+        {
+            ServerError,
+            ConnectionTimedOut,
+            NetworkIssues
+        }
+
+        public ITry<T, NetworkOperationError> PerformNetworkOperation<T>(Func<Unit, T> operation)
+        {
+            var connectionErrorStatuses = new List<WebExceptionStatus>
+            {
+                WebExceptionStatus.ConnectFailure,
+                WebExceptionStatus.ConnectionClosed,
+                WebExceptionStatus.SecureChannelFailure,
+                WebExceptionStatus.NameResolutionFailure
+            };
+
+            try
+            {
+                return Try.Success<T, NetworkOperationError>(operation(Unit.Value));
+            }
+            catch (TaskCanceledException)
+            {
+                return Try.Error<T, NetworkOperationError>(NetworkOperationError.ConnectionTimedOut);
+            }
+            catch (WebException e) when (e.Response.As<HttpWebResponse>().Map(r => (int)r.StatusCode > 500 && (int)r.StatusCode < 599).GetOrFalse())
+            {
+                return Try.Error<T, NetworkOperationError>(NetworkOperationError.ServerError);
+            }
+            catch (WebException e) when (connectionErrorStatuses.Contains(e.Status))
+            {
+                return Try.Error<T, NetworkOperationError>(NetworkOperationError.NetworkIssues);
+            }
+            // Notice that general Exception is not handled. We're handling the exceptions we expect, so the signature makes sense.
+            // The purpose of ITry is not covering every exception. But that methods should show what are the expected outputs and make the call site handle all of them.
+        }
+
+        private void TransformingResultValuesWithMap(decimal number, decimal multiplier)
+        {
+            ITry<decimal, NetworkOperationError> multiplicationResult = PerformNetworkOperation(_ => number * multiplier);
+            ITry<int, NetworkOperationError> flooredMultiplicationResult = multiplicationResult.Map(r => (int)r);
+            ITry<string, NetworkOperationError> stringifiedDivisionResult = multiplicationResult.Map(r => r.ToString());
```

Comment:
Multiplication as network operation looks weird. I'd be afraid of "killing the example" by it not making sense.